pandas/io/excel/_base.py

```diff
@@ -666,6 +666,17 @@ class ExcelWriter(metaclass=abc.ABCMeta):
         be parsed by ``fsspec``, e.g., starting "s3://", "gcs://".
 
         .. versionadded:: 1.2.0
+    if_sheet_exists : {'new', 'replace', 'overwrite', 'fail'}, default 'new'
+        How to behave when trying to write to a sheet that already
+        exists (append mode only).
+
+        * new: Create a new sheet, with a name determined by the engine.
+        * replace: Delete the contents of the sheet before writing to it.
+        * overwrite: Write directly to the named sheet
```

Comment:
What is the use case for overwrite?

Reply:
Published statistics, at least in the UK, often have sheets which combine headings, date, description, data quality notes, etc with data tables. To automate something like this you would probably have a pre-written template and then write your data from pandas into specific sheets at specific locations.

An example I happened to be looking at recently, [England's daily, weekly and monthly vaccination figures](https://www.england.nhs.uk/statistics/statistical-work-areas/covid-19-vaccinations/).

Reply:
@mrob95 - does the current implementation in this PR overwrite the template formatting? E.g. if a column in the template is formatted to percent and I have a DataFrame with 0.5, will it be displayed in excel as 50%?

Reply:
@rhshadrach It doesn't overwrite cell formatting unless there is an alternative style set. E.g. `df.style.set_properties(**{"number-format": "0.00%"})` will overwrite number formatting but otherwise the written cells inherit the previous formatting, including conditional formatting.

The only exception to this I can see is headers and indexes, which have a hardcoded style which I think will always overwrite certain formats (see `def header_style` in `io/formats/excel.py`). This may not be ideal for certain use cases but seems like a separate issue, about which there is already discussion (https://github.com/pandas-dev/pandas/issues/25185).

Between this and [pandas own excel formatting options](https://pandas.pydata.org/pandas-docs/stable/user_guide/style.html#Export-to-Excel) I think the options are pretty good for styling tables written using `if_sheet_exists="overwrite"`.